waiter/src/waiter/process_request.clj

```diff
@@ -658,7 +658,7 @@
   (if trailers
     (let [correlation-id (cid/get-correlation-id)
           trailers-copy-ch (async/chan 1)
-          {:keys [grpc-status] :as grpc-headers} (select-keys headers ["grpc-message" "grpc-status"])]
+          {:strs [grpc-status] :as grpc-headers} (select-keys headers ["grpc-message" "grpc-status"])]
```

Comment:
The `num-messages=0` test fails without this fix.

Reply:
the original looks like a bug too. `grpc-status` will be nil because the second argument: `["grpc-message" "grpc-status"]` will cause the resulting map to never have the `:grpc-status` defined

Reply:
Yeah, that was the bug :(